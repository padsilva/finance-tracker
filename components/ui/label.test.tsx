import { render } from "@testing-library/react";

import { Label } from "./label";

describe("Label Component", () => {
  it("should render default label correctly", () => {
    const { container } = render(<Label>Default Label</Label>);

    expect(container).toMatchSnapshot();
  });

  it("should render with custom className", () => {
    const { container } = render(
      <Label className="custom-class">Custom Label</Label>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with htmlFor attribute", () => {
    const { container } = render(
      <div>
        <Label htmlFor="test-input">Input Label</Label>
        <input id="test-input" type="text" />
      </div>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with disabled state", () => {
    const { container } = render(
      <div>
        <Label>
          <input type="checkbox" disabled />
          <span>Disabled checkbox label</span>
        </Label>
      </div>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with required state", () => {
    const { container } = render(
      <div>
        <Label>
          Required field
          <span className="ml-1 text-red-500">*</span>
        </Label>
      </div>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with different variants", () => {
    const { container } = render(
      <>
        <Label className="text-sm">Small Label</Label>
        <Label className="text-base">Medium Label</Label>
        <Label className="text-lg">Large Label</Label>
        <Label className="font-bold">Bold Label</Label>
        <Label className="italic">Italic Label</Label>
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with ARIA attributes", () => {
    const { container } = render(
      <Label htmlFor="test-input" aria-label="Test label" aria-required="true">
        ARIA Label
      </Label>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with data attributes", () => {
    const { container } = render(
      <Label data-testid="test-label" data-cy="cypress-label">
        Test Label
      </Label>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with nested content", () => {
    const { container } = render(
      <Label>
        <span className="font-bold">Profile</span>
        <span className="ml-2 text-gray-500">(Optional)</span>
      </Label>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render in form context", () => {
    const { container } = render(
      <div className="space-y-2">
        <Label htmlFor="username">Username</Label>
        <input
          id="username"
          type="text"
          className="peer"
          placeholder="Enter username"
        />
        <Label htmlFor="email">Email</Label>
        <input
          id="email"
          type="email"
          className="peer"
          placeholder="Enter email"
        />
      </div>,
    );

    expect(container).toMatchSnapshot();
  });
});
