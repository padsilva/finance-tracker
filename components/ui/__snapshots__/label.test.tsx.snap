// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component should render default label correctly 1`] = `
<div>
  <label
    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
  >
    Default Label
  </label>
</div>
`;

exports[`Label Component should render in form context 1`] = `
<div>
  <div
    class="space-y-2"
  >
    <label
      class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
      for="username"
    >
      Username
    </label>
    <input
      class="peer"
      id="username"
      placeholder="Enter username"
      type="text"
    />
    <label
      class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
      for="email"
    >
      Email
    </label>
    <input
      class="peer"
      id="email"
      placeholder="Enter email"
      type="email"
    />
  </div>
</div>
`;

exports[`Label Component should render with ARIA attributes 1`] = `
<div>
  <label
    aria-label="Test label"
    aria-required="true"
    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
    for="test-input"
  >
    ARIA Label
  </label>
</div>
`;

exports[`Label Component should render with custom className 1`] = `
<div>
  <label
    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 custom-class"
  >
    Custom Label
  </label>
</div>
`;

exports[`Label Component should render with data attributes 1`] = `
<div>
  <label
    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
    data-cy="cypress-label"
    data-testid="test-label"
  >
    Test Label
  </label>
</div>
`;

exports[`Label Component should render with different variants 1`] = `
<div>
  <label
    class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-sm"
  >
    Small Label
  </label>
  <label
    class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-base"
  >
    Medium Label
  </label>
  <label
    class="font-medium peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-lg"
  >
    Large Label
  </label>
  <label
    class="text-sm leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-bold"
  >
    Bold Label
  </label>
  <label
    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 italic"
  >
    Italic Label
  </label>
</div>
`;

exports[`Label Component should render with disabled state 1`] = `
<div>
  <div>
    <label
      class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
    >
      <input
        disabled=""
        type="checkbox"
      />
      <span>
        Disabled checkbox label
      </span>
    </label>
  </div>
</div>
`;

exports[`Label Component should render with htmlFor attribute 1`] = `
<div>
  <div>
    <label
      class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
      for="test-input"
    >
      Input Label
    </label>
    <input
      id="test-input"
      type="text"
    />
  </div>
</div>
`;

exports[`Label Component should render with nested content 1`] = `
<div>
  <label
    class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
  >
    <span
      class="font-bold"
    >
      Profile
    </span>
    <span
      class="ml-2 text-gray-500"
    >
      (Optional)
    </span>
  </label>
</div>
`;

exports[`Label Component should render with required state 1`] = `
<div>
  <div>
    <label
      class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
    >
      Required field
      <span
        class="ml-1 text-red-500"
      >
        *
      </span>
    </label>
  </div>
</div>
`;
